### ISSUE TITLE
Odd looking SelectEntry with long PlaceHolder - overlap by dropdown ActionItem

### ISSUE THREAD

**Issue by sdassow** [fyne-io/fyne#4430]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Using SelectEntry with a long PlaceHolder looks odd because the ActionItem to open the dropdown is rendered partially over the PlaceHolder text.

### How to reproduce

1. Run example code
2. Hover with mouse over ActionItem to open dropdown
3. See overlap as shown in screenshot

### Screenshots

<img width="112" alt="selectentry-placeholder-overlap" src="https://github.com/fyne-io/fyne/assets/70379952/0e64d093-b21b-4983-b155-0d0ea718bf1d">

### Example code

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("PlaceHolder SelectEntry Overlap Test")

	e := widget.NewSelectEntry([]string{})
	e.PlaceHolder = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."

	w.SetContent(container.NewCenter(e))
	w.ShowAndRun()
}
```

### Fyne version

2.4.2

### Go compiler version

different versions

### Operating system and version

various deskop oses (not tested on mobile)

**Comment by andydotxyz** [comment]
OK I am working on this and realised that perhaps we were mistaken in making those buttons so small anyway. So here is the result of what I think is possibly a better arrangement. It's much more space than before but keen for feedback before I hope a PR.

![Screenshot 2024-04-09 at 15 21 48](https://github.com/fyne-io/fyne/assets/294436/dbdc4e63-f4e8-4f5a-92be-4a6cec3cbff0)
![Screenshot 2024-04-09 at 15 22 13](https://github.com/fyne-io/fyne/assets/294436/932de6d6-f7e4-4318-9741-0c1bed3836e9)
![Screenshot 2024-04-09 at 15 23 48](https://github.com/fyne-io/fyne/assets/294436/4e597051-6ebe-46a1-a268-baf7793f15fb)

This is the 3 possible combinations of accessory and validation that a (Select)Entry can have.
Note of course the buttons are hovered to show the extents now are approx min-size instead of icon size.

**Comment by sdassow** [comment]
Makes absolute sense to me and looks good as well.

**Comment by andydotxyz** [comment]
Sorry for the delay, there is finally a PR open with that code!

**Comment by andydotxyz** [comment]
Fixed on `develop` for v2.6.0.
As noted on the PR this was deemed a behaviour change so won't be in the 2.5.x release series.
